gog/graph.go

```diff
@@ -47,6 +47,16 @@ type Grapher struct {
 
 	seenDocObjs map[types.Object]struct{}
 	seenDocKeys map[string]struct{}
+
+	// pkgDefs accumulates the defs from the package being graphed currently in Graph().
+	// If the package is graphed successfully, the defs are added to Output.
+	pkgDefs []*Def
```

Comment:
I added this mainly because if graphing fails midway on a package, we may have partial data from that package. That seems OK but may lead to inconsistencies downstream, and this preserves the behavior from graphing one unit at a time, i.e. if a unit fails then no data for that unit is returned.

But I'm not too sure about this. We could change the behavior to return partial graph data from failed units if there is no obvious issue in that (other than diverging in behavior from other toolchains).